src/sql/workbench/services/notebook/common/sqlSessionManager.ts

```diff
@@ -299,7 +299,24 @@ export class SQLFuture extends Disposable implements FutureInternal {
 	setIOPubHandler(handler: nb.MessageHandler<nb.IIOPubMessage>): void {
 		this._register(this._queryRunner.onBatchEnd(batch => {
 			this._queryRunner.getQueryRows(0, batch.resultSetSummaries[0].rowCount, 0, 0).then(d => {
-				let data: SQLData = {
+				let columns = batch.resultSetSummaries[0].columnInfo.map(c => c.columnName);
```

Comment:
Please refactor to 2 functions `convertToDataResource` and `convertToHtmlTable`. We may want to use these elsewhere, plus it's easier to understand example what each part of this does.

Reply:
Also maybe a stretch but... unit tests??? 😄